### PR TITLE
feat: Update ListItem Loading State into Shimmering loader

### PR DESCRIPTION
**Pull request by Golodhros**
### Summary of Changes
Adds a new Resource Shimmering component
Uses it on the homepage for the My Bookmarks and Popular Tables sections
Uses it on the Search Page

Amundsen_-_Data_Discovery_Portal

### Tests
Updated tests and cleaned up code

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes, including screenshots of any UI changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/components/SearchPage/index.tsx [hunk 1]
```diff
@@ -157,8 +160,9 @@ export class SearchPage extends React.Component<SearchPageProps> {
 
   renderContent = () => {
     if (this.props.isLoading) {
-      return <LoadingSpinner />;
+      return <ShimmeringResourceLoader numItems={NUMBER_OF_FAKE_RESULTS} />;
```
[the remark this answers is not in the document]
Comment (Golodhros) [reply amundsen_application/static/js/components/SearchPage/index.tsx:163]: Makes sense, will use that variable.

File: amundsen_application/static/js/components/common/Bookmark/MyBookmarks/index.tsx [hunk 2]
```diff
@@ -73,17 +78,21 @@ export class MyBookmarks extends React.Component<MyBookmarksProps> {
   };
 
   render() {
-    if (!this.props.isLoaded) {
-      return null;
-    }
+    let content = <ShimmeringResourceLoader />;
```
[the remark this answers is not in the document]
Comment (Golodhros) [reply amundsen_application/static/js/components/common/Bookmark/MyBookmarks/index.tsx:81]: good point, adding that.

File: amundsen_application/static/js/components/common/PopularTables/index.tsx [hunk 3]
```diff
@@ -36,24 +37,34 @@ export class PopularTables extends React.Component<PopularTablesProps> {
   }
 
   render() {
+    const { popularTables, isLoaded } = this.props;
+    let content = <ShimmeringResourceLoader />;
```
[the remark this answers is not in the document]
Comment (Golodhros) [reply amundsen_application/static/js/components/common/PopularTables/index.tsx:41]: Will do!

File: amundsen_application/static/js/components/common/PopularTables/index.tsx [hunk 4]
```diff
@@ -20,8 +18,11 @@ import {
   POPULAR_TABLES_PER_PAGE,
 } from './constants';
 
+import './styles.scss';
+
 export interface StateFromProps {
   popularTables: TableResource[];
+  isLoaded: boolean;
```
[the remark this answers is not in the document]
Comment (Golodhros) [reply amundsen_application/static/js/components/common/PopularTables/index.tsx:25]: Oh, I see.
I took as a template the code on MyBookmarks, which has:
```
{
    myBookmarks: state.bookmarks.myBookmarks,
    isLoaded: state.bookmarks.myBookmarksIsLoaded
}
```

Do you want me to change it in both?

File: amundsen_application/static/js/ducks/popularTables/reducer.ts [hunk 5]
```diff
@@ -20,18 +20,33 @@ export function getPopularTablesSuccess(
...
-export type PopularTablesReducerState = TableResource[];
+export interface PopularTablesReducerState {
+  popularTables: TableResource[];
+  popularTablesIsLoaded: boolean;
+}
 
-const initialState: PopularTablesReducerState = [];
+const initialState: PopularTablesReducerState = {
+  popularTables: [],
+  popularTablesIsLoaded: false,
```
[the remark this answers is not in the document]
Comment (Golodhros) [reply amundsen_application/static/js/ducks/popularTables/reducer.ts:30]: I see. It wasn't that bad to add, I was taking the Bookmarks feature as a template.

File: amundsen_application/static/js/components/common/PopularTables/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (ttannis) [reply amundsen_application/static/js/components/common/PopularTables/index.tsx:25]: I don't feel too strongly about changing anything at the moment, would rather invest the energy someday into just improving how we store the loading states on the globalState.

**Pull request merged**